### PR TITLE
Added a checkbox to hide invisible objects in collections

### PR DESCRIPTION
**Pull request by gauravsharma21**
With reference to issue #378, I added a checkbox to hide invisible objects in collections

### Preview
Hnet-image

### CODE REVIEW

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.scss [hunk 1]
```diff
@@ -4,11 +4,17 @@
...
     height: 15%;
     min-height: 2rem;
+    justify-content: space-between;
 
     span {
       color: var(--phoenix-text-color-secondary);
       margin-right: 1rem;
     }
+
+    .checkbox-wrapper {
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.scss:14]: The layout you have can be achieved with flex. No need to use any other properties.

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html [hunk 2]
```diff
@@ -11,19 +11,26 @@
...
+          >
+            <option value="" selected disabled hidden>Choose Collection</option>
+            <option *ngFor="let collection of collections" [value]="collection">
+              {{ collection }}
+            </option>
+          </select>
+        </div>
+      </div>
+      <div class="checkbox-wrapper" *ngIf="showingCollection">
+        <mat-checkbox (change)="toggleInvisible($event.checked)">
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html:31]: No need for a wrapper here. Once you use flex the right way, the item will become right aligned.

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html [hunk 3]
```diff
@@ -11,19 +11,26 @@
       class="collectionSelector mb-2 d-flex align-items-center"
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html:11]: Use this wrapper to achieve the current layout you have using bootstrap's flex related classes.

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.ts [hunk 4]
```diff
@@ -13,9 +13,12 @@ import { EventDisplayService } from '../../../../services/event-display.service'
 })
 export class CollectionsInfoOverlayComponent implements OnInit {
   @Input() showObjectsInfo: boolean;
+  hideInvisible: boolean;
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.ts:16]: Just having this `hideInvisible` for keeping the checkbox value is enough. No need to make any other changes to the component code, you could just use the `isCut` of an object property to decide whether to display a row or not.

**Review by 9inpachi — CHANGES_REQUESTED** [review]
Would recommend to place the checkbox below the collection select or right next to it. The way it is now, if the collections info panel has a large width, the checkbox isn't as prominent in terms of UX and can be missed.

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html [hunk 5]
```diff
@@ -50,7 +57,8 @@
             [ngClass]="{
               'active-object':
                 activeObject && activeObject.value === object.uuid,
-              'is-cut': object.isCut
+              'is-cut': object.isCut,
+              hide: object.isCut && hideInvisible
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html:61]: Bootstrap already has a class for this: `d-none`.

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html [hunk 6]
```diff
@@ -11,19 +11,26 @@
...
+            name="collection"
+            (change)="changeCollection($event.target.value)"
+          >
+            <option value="" selected disabled hidden>Choose Collection</option>
+            <option *ngFor="let collection of collections" [value]="collection">
+              {{ collection }}
+            </option>
+          </select>
+        </div>
+        <div class="checkbox-wrapper" *ngIf="showingCollection">
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html:29]: This wrapper looks redundant to me. I am not sure but does the layout not work without this?

File: packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.scss [hunk 7]
```diff
@@ -9,6 +9,15 @@
...
+      color: var(--phoenix-text-color-secondary);
```
Comment (9inpachi) [inline packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.scss:14]: It should pick up the primary color automatically. which should be fine.

**Review by 9inpachi — CHANGES_REQUESTED** [review]

**Comment by gauravsharma21** [comment]
image

@9inpachi Is this fine or should I add some margin between the dropdown & checkbox?

**Comment by EdwardMoyse** [comment]
> image
> 
> @9inpachi Is this fine or should I add some margin between the dropdown & checkbox?

I think some margin would look better.

**Review by 9inpachi — APPROVED** [review]
Thanks a lot!

**Pull request merged**